python/lsst/daf/butler/core/formatter.py

```diff
@@ -58,22 +59,34 @@ class Formatter(metaclass=ABCMeta):
         Identifies the file to read or write, and the associated storage
         classes and parameter information.  Its value can be `None` if the
         caller will never call `Formatter.read` or `Formatter.write`.
+    dataId : `DataCoordinate`, optional
+        Data ID associated with this formatter.
```

Comment:
I realize only the docs are new, but I'm curious why `Formatter` wants a data ID.

Reply:
Because DECam raw files have no other way of knowing which detector you actually wanted the formatter to read.